proposal/src/main/scala/explore/proposal/ProposalDetailsEditor.scala

```diff
@@ -26,11 +32,90 @@ final case class ProposalDetailsEditor(proposalDetails: View[ProposalDetails])
 object ProposalDetailsEditor {
   type Props = ProposalDetailsEditor
 
-  val component =
-    ScalaComponent
-      .builder[Props]
-      .render_P { props =>
-        val details = props.proposalDetails
+  @Lenses
+  final case class State(showPartnerSplitsModal: Boolean, splits: List[PartnerSplit])
+
+  implicit val propsReuse: Reusability[Props] = Reusability.derive
+  implicit val stateReuse: Reusability[State] = Reusability.derive
+
+  private def formatTime(time: Double) = f"$time%.2fh"
+
+  private def partnerSplits(splits: List[PartnerSplit]): TagMod = {
+    val ps = splits
+      .sortBy(_.percent)(Ordering[Int].reverse)
+      .toTagMod(ps => partnerSplit(ps))
+    <.div(ps, ExploreStyles.FlexContainer, ExploreStyles.FlexWrap)
+  }
+
+  private def partnerSplit(ps: PartnerSplit): TagMod = {
+    val id   = s"${ps.partner.tag}-split"
+    val text = f"${ps.percent}%%"
+    partnerSplitData(ps.partner, id, text)
+  }
+
+  // switch to use partner.shortName after next lucuma-core release
+  private def partnerSplitData(partner: Partner, id: String, data: String) = {
+    val img: TagMod  =
+      <.img(^.src := PartnerFlags.smallFlag(partner),
+            ^.alt := s"${partner.name}  Flag",
+            ExploreStyles.PartnerSplitFlag
+      )
+    val span: TagMod = <.span(data)
+
+    FormStaticData(id = id, value = TagMod(img, span), label = partner.name)(
+      ExploreStyles.FlexShrink(0)
+    )
+  }
+
+  private def bandSplits(splits: List[PartnerSplit], total: Double): TagMod =
+    splits match {
+      case Nil => TagMod.empty
+      case _   =>
+        <.span(
+          splits
+            .sortBy(_.percent)(Ordering[Int].reverse)
+            .map(bandSplit(_, total))
+            .mkString("(", ", ", ")"),
+          ExploreStyles.TextInForm
+        )
+    }
+
+  private def bandSplit(ps: PartnerSplit, total: Double) = {
+    // switch to ps.abbreviation when available in next lucuma-core release
+    val partnerText = ps.partner.tag.toUpperCase
+    val timeText    = formatTime(total * ps.percent / 100)
+    s"$partnerText $timeText"
+  }
+
+  // temporary until abbreviation is in Partner
+  class Backend($ : BackendScope[Props, State]) {
+
+    def render(props: Props, state: State) = {
+      val details        = props.proposalDetails
+      val band1And2Hours = details.zoom(ProposalDetails.band1And2Hours).get
+      val band3Hours     = details.zoom(ProposalDetails.band3Hours).get
+
+      def updateStateSplits(splits: List[PartnerSplit]): Callback =
+        $.modState(State.splits.set(splits))
```

Comment:
not very important, here you can also use `$.modStateL(State.splits)(splits)`. It doesn't make much difference here but it is nicer for nested state items

Reply:
That's a good thing to know about. So many little hidden bits to find. :) In this case, it seems I need `setStateL`, though.